include/wx/generic/maskededit.h

```diff
@@ -0,0 +1,681 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        wx/generic/maskededit.h
+// Purpose:     wxMaskedEdit interface
+// Author:      Manuel Martin
+// Modified by:
+// Created:     2012-07-30
+// RCS-ID:      $Id$
+// Copyright:   (c)
+// Licence:     wxWindows license
+/////////////////////////////////////////////////////////////////////////////
+
+#ifndef _WX_MASKED_EDIT_H_
+#define _WX_MASKED_EDIT_H_
+
+#include "wx/defs.h"
+
+#if wxUSE_MASKED_EDIT
+
+#include "wx/colour.h"
+#include "wx/vector.h"
+
+#if wxUSE_TEXTCTRL
+    #include "wx/textctrl.h"
+#endif // wxUSE_TEXTCTRL
+
+#if wxUSE_COMBOBOX
+    #include "wx/combobox.h"
+#endif // wxUSE_COMBOBOX
+
+
+//Colours used for different cases.
+struct WXDLLIMPEXP_ADV wxMaskedEditColours
+{
+    //When everything is OK
+    wxColour colOKForegn;
+    wxColour colOKBackgn;
+    //When something is wrong
+    wxColour colWrForegn;
+    wxColour colWrBackgn;
+    //When nothing is filled, empty control's value
+    wxColour colEmForegn;
+    wxColour colEmBackgn;
+};
+
+//An object to store flags
+class WXDLLIMPEXP_ADV wxEditFieldFlags
+{
+public:
+    wxEditFieldFlags(wxAlignment alignment = wxALIGN_LEFT)
+    {
+        //Only left or right alignment
+        m_alignment = alignment == wxALIGN_RIGHT ? wxALIGN_RIGHT : wxALIGN_LEFT;
+        m_fillChar = wxChar(' ');
+        m_paddingChar = wxChar('\0');
+    }
+
+    //Setters for all flags. Return the object itself, so that calls to them
+    // can be chained
+
+    wxEditFieldFlags& SetAlignment(wxAlignment alignment)
+    {
+        //Only left or right alignment
+        m_alignment = alignment == wxALIGN_RIGHT ? wxALIGN_RIGHT : wxALIGN_LEFT;
+        return *this;
+    }
+
+    wxEditFieldFlags& SetFillChar(wxChar fillchar)
+    {
+        if ( !wxIsprint(fillchar) )
+        {
+            wxFAIL_MSG("Warning: Unprintable character in masked flags");
+        }
+        m_fillChar = fillchar;
+        return *this;
+    }
+
+    wxEditFieldFlags& SetPaddingChar(wxChar paddingchar)
+    {
+        if ( paddingchar != '\0' && !wxIsprint(paddingchar) )
```

Comment:
`paddingchar != '\0'` check should not be needed, `wxIsprint()` would return `false` for `\0`.